src/librustdoc/visit.rs

```diff
@@ -0,0 +1,71 @@
+use crate::clean::*;
+
+crate trait DocVisitor: Sized {
+    fn visit_item(&mut self, item: &Item) {
+        self.visit_item_recur(item)
+    }
+
+    /// don't override!
```

Comment:
Like how TypeFoldable provides this method for types, could an extension trait be attached to ItemKind to take care of this?

Reply:
Hmm, could you elaborate? I'm not sure why that would be better. I'm generally not a fan of too many extension traits since they add complexity.

Reply:
If you don't like the idea, then it's fine, let's not do that. I brought it up because it would enforce the "don't override" rule instead of just making it a comment.

Reply:
I think I see what you mean. We could do this without an extension trait since `ItemKind` is local. But I'm not sure if the extra complexity of not having all the visitor code in one place is worth it. Also, the current way `DocVisitor` (and `DocFolder`) works is consistent with similar visitors in rustc (like `mir::Visitor`). But I'd still be open to us trying it.